### PR TITLE
WIP Feature/cert pipeline

### PR DESCRIPTION
**Pull request by dacleyra**
To pass certification checks
- annotations.yaml needs
  - openshift.versions
	- CHANNELS, DEFAULT_CHANNELS with default; must be set to generate their required associated annotations
- remove the operator service account yaml, this should be created by OLM & fails a cert check
  - didn't seem to be a great way to remove it via a kustomize option; make bundle vs. make install needing the sa

### CODE REVIEW

File: v2/hack/certify/certify.sh [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+# kustomize questionable capability to remove the service account
+rm -Rf  $TMP_DIR/certified-operators-preprod/operators/redhat-marketplace-operator/$VERSION/manifests/redhat-marketplace-operator_v1_serviceaccount.yaml
+
+echo "organization: redhat-marketplace" > $TMP_DIR/certified-operators-preprod/config.yaml
+echo "cert_project_id: 5f68c9457115dbd1183ccab6" > $TMP_DIR/certified-operators-preprod/operators/redhat-marketplace-operator/ci.yaml
+
+cd $TMP_DIR/certified-operators-preprod
+git add --all
+git commit -m $VERSION
+git push
```
Comment (zach-source) [inline v2/hack/certify/certify.sh:146]: Can you use branches or it has to be a main commit&push?

File: v2/hack/certify/certify.sh [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+	unset PHASE
+	until [ "$PHASE" == "Complete" ]
+	do
+		PHASE=$(oc get installplan $INSTALL_PLAN -n $2 --output=jsonpath={.status.phase})
+    if [ "$PHASE" == "Failed" ]; then
+      set +x
+      sleep 3
+      echo "InstallPlan $INSTALL_PLAN for subscription $1 failed."
+      echo "To investigate the reason of the InstallPlan failure run:"
+      echo "oc describe installplan $INSTALL_PLAN -n $2"
```
Comment (zach-source) [inline v2/hack/certify/certify.sh:40]: Nice comment!

File: v2/hack/certify/certify.sh [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+cd $TMP_DIR/certified-operators-preprod
+git add --all
+git commit -m $VERSION
+git push
+
+
+# Run the Pipeline
+
+cd $TMP_DIR/operator-pipelines
+curl https://mirror.openshift.com/pub/openshift-v4/clients/pipeline/0.19.1/tkn-linux-amd64-0.19.1.tar.gz | tar -xz
```
Comment (zach-source) [inline v2/hack/certify/certify.sh:152]: Add a check for a local tkn or one on the path. Prefer local if it's there. Print version.

File: v2/hack/certify/certify.sh [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+
+# Run the Pipeline
+
+cd $TMP_DIR/operator-pipelines
+curl https://mirror.openshift.com/pub/openshift-v4/clients/pipeline/0.19.1/tkn-linux-amd64-0.19.1.tar.gz | tar -xz 
+
+GIT_REPO_URL=https://github.com/redhat-marketplace/certified-operators-preprod.git
+BUNDLE_PATH=operators/redhat-marketplace-operator/$VERSION
+
+./tkn pipeline start operator-ci-pipeline \
```
Comment (zach-source) [inline v2/hack/certify/certify.sh:157]: Does the pipeline clean up after it's done?

File: v2/hack/certify/certify.sh [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+  --param git_repo_url=$GIT_REPO_URL \
+  --param git_branch=stage \
+  --param bundle_path=$BUNDLE_PATH \
+  --param env=stage \
+  --workspace name=pipeline,volumeClaimTemplateFile=templates/workspace-template.yml \
```
Comment (zach-source) [inline v2/hack/certify/certify.sh:162]: Does there need to exist a template/workspace-template.yml somewhere? Or is this in the other repo?

**Review by zach-source — CHANGES_REQUESTED** [review]
Minor nits.

File: v2/hack/certify/certify.sh (hunk 1 above)
Comment (dacleyra) [reply v2/hack/certify/certify.sh:146]: they actually just changed the repo in the instructions, and the default branch was stage, now it is main
we can set this to our choice of branch, the git_branch is specified in the tekton run
let me know how you want to handle it

File: v2/hack/certify/certify.sh (hunk 2 above)
Comment (dacleyra) [reply v2/hack/certify/certify.sh:40]: checksub routine was written back in pre-ICPA days, before there was a product and ansible installer, and OLM was a little rougher around the edges

this at least got people to not immediately ask me blindly why a hacked up pre-prod install didn't work...

File: v2/hack/certify/certify.sh (hunk 4 above)
Comment (dacleyra) [reply v2/hack/certify/certify.sh:157]: pipeline will delete created artifacts whether the pipeline succeeded for failed
however, this is where we got into the failure related to the stuck finalizer problem, as a result of installing to our non default namespace

File: v2/hack/certify/certify.sh (hunk 1 above)
Comment (zach-source) [reply v2/hack/certify/certify.sh:146]: Let's create a branch named for the release off the version and pass that. It'll let us keep stuff separate. Maybe at some point we can trigger actions on merge. If that's too complicated to do right now we can do follow on. First priority is a release of 2.4.0

File: v2/hack/certify/certify.sh (hunk 3 above)
Comment (dacleyra) [reply v2/hack/certify/certify.sh:152]: this is all being done in a tmp directory, tkn won't exist run-to-run, unless you want it installed elsewhere

File: v2/hack/certify/certify.sh (hunk 5 above)
Comment (dacleyra) [reply v2/hack/certify/certify.sh:162]: At this point in the script, we are in the operator-pipelines dir, so the file is

https://github.com/redhat-openshift-ecosystem/operator-pipelines/blob/main/templates/workspace-template.yml

File: v2/hack/certify/certify.sh (hunk 3 above)
Comment (zach-source) [reply v2/hack/certify/certify.sh:152]: Ah, missed that. Carry on.

File: v2/hack/certify/certify.sh (hunk 2 above)
Comment (zach-source) [reply v2/hack/certify/certify.sh:40]: Hah - it's the little things tho.

**Review by zach-source — CHANGES_REQUESTED** [review]
Still WIP

**Pull request merged**